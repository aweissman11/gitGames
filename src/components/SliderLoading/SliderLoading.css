.slider-loading {
  height: 100%;
  width: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
}

.slider-loading-text {
  display: flex;
  font-size: 2em;
  justify-content: center;
  align-items: center;
  color: #d8daeb;
}

.slider-loading-spinner {
  margin-right: 15px;
}