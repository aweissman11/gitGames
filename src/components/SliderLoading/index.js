import React from 'react';
import ReactLoading from 'react-loading';

import './SliderLoading.css';

export const SliderLoading = (props) => (
  <div className='slider-loading'>
    <h4 className='slider-loading-text'>
    <span>
      <ReactLoading className='slider-loading-spinner' type={'spokes'} color={'#b7b7b7'} height={40} width={40} />
    </span>
    {props.msg}
    </h4>
  </div>
);

export default SliderLoading;