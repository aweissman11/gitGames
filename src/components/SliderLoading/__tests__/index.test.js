import React from 'react';
import { shallow } from 'enzyme';

import SliderLoading from '../index';

describe('SliderLoading', () => {

  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<SliderLoading />)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
  
})
