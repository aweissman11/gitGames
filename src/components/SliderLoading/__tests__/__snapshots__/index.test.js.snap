// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SliderLoading should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SliderLoading />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <h4
        className="slider-loading-text"
      >
        <span>
          <Loading
            className="slider-loading-spinner"
            color="#b7b7b7"
            delay={0}
            height={40}
            type="spokes"
            width={40}
          />
        </span>
      </h4>,
      "className": "slider-loading",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span>
            <Loading
              className="slider-loading-spinner"
              color="#b7b7b7"
              delay={0}
              height={40}
              type="spokes"
              width={40}
            />
          </span>,
          undefined,
        ],
        "className": "slider-loading-text",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Loading
              className="slider-loading-spinner"
              color="#b7b7b7"
              delay={0}
              height={40}
              type="spokes"
              width={40}
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": "slider-loading-spinner",
              "color": "#b7b7b7",
              "delay": 0,
              "height": 40,
              "type": "spokes",
              "width": 40,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "span",
        },
        undefined,
      ],
      "type": "h4",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <h4
          className="slider-loading-text"
        >
          <span>
            <Loading
              className="slider-loading-spinner"
              color="#b7b7b7"
              delay={0}
              height={40}
              type="spokes"
              width={40}
            />
          </span>
        </h4>,
        "className": "slider-loading",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span>
              <Loading
                className="slider-loading-spinner"
                color="#b7b7b7"
                delay={0}
                height={40}
                type="spokes"
                width={40}
              />
            </span>,
            undefined,
          ],
          "className": "slider-loading-text",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Loading
                className="slider-loading-spinner"
                color="#b7b7b7"
                delay={0}
                height={40}
                type="spokes"
                width={40}
              />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "className": "slider-loading-spinner",
                "color": "#b7b7b7",
                "delay": 0,
                "height": 40,
                "type": "spokes",
                "width": 40,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "span",
          },
          undefined,
        ],
        "type": "h4",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
